src/components/dashboard/Transactions.vue

```diff
@@ -0,0 +1,173 @@
+<script setup lang="ts">
+import { useConfirm } from "primevue/useconfirm";
+import { useToast } from "primevue/usetoast";
+import {onMounted, ref} from "vue";
+import TransactionDialog from "./TransactionDialog.vue";
+import { useTransactionStore } from '@/stores/transaction';
+
+const confirm = useConfirm();
+const toast = useToast();
+const transaction = useTransactionStore()
+const id = ref<number | undefined>();
+
+const confirmDelete = (id : number) => {
+	confirm.require({
+		message: 'Do you want to delete this record?',
+		header: 'Danger Zone',
+		icon: 'pi pi-info-circle',
+		rejectLabel: 'Cancel',
+		acceptLabel: 'Delete',
+		rejectClass: 'p-button-secondary p-button-outlined',
+		acceptClass: 'p-button-danger',
+		accept: () => {
+			toast.add({ severity: 'error', summary: 'Confirmed', detail: 'Record deleted', life: 3000 });
+			transaction.deleteTransaction(id)
+		},
+	});
+};
+
+onMounted(() => {
+	transaction.setTransactions();
+})
+
+</script>
+
+<template>
+	<Toast />
+	<ConfirmDialog></ConfirmDialog>
+	<TransactionDialog :id='id'/>
+	<div class="dashboard-header">
+		<div class="title">
+		Transaction History
+		</div>
+		<Button label="Add Transaction" icon="pi pi-plus" @click="transaction.toggleVisibility(); id = undefined;" />
```

Comment:
Bakit undefined yung naka-assign kay id dito?

Reply:
Para walang id na mapasa sa child component. Kasi yung condition don is if id === undefined then add transaction siya. Tangalin ko nalang ba? Since default value naman ata ng id is undefined

Reply:
ayyy reassignment pala to. Kasi kapag pinindot yung update, nakakaroon ng value si ID. So, kapag pinindot si add, dapat bumalik sa undefined yung value niya